### PR TITLE
version 0.7 RC

### PR DESCRIPTION
**Pull request by alexcb**
New 0.7 version with corresponding default-enabled features, along with
updated tests.

### CODE REVIEW

**Comment by alexcb** [comment]
There are a few tests that are left at `VERSION 0.6`, which I will address in follow-up PRs.

File: examples/secrets/Earthfile [hunk 1]
```diff
@@ -1,3 +1,4 @@
+VERSION 0.6
```
Comment (nelsam) [inline examples/secrets/Earthfile:1]: Just double-checking that this was intentional.

A comment explaining why this example uses `0.6` would be useful here. I think if I were trying to use earthly and was browsing through the examples, I would be surprised to see old versions pinned with no explanation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -96,6 +125,10 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
 
 - Loading Docker images as part of `WITH DOCKER` is now faster through the use of an embedded registry in Buildkit. This functionality was previously hidden (`VERSION --use-registry-for-with-docker`) and was only auto-enabled for Earthly Satellite users. It is now enabled by default for all builds. [#1268](https://github.com/earthly/earthly/issues/1268)
 
+### Changed
+
+- `VERSION` command is now required.
+
```
Comment (vladaionescu) [inline CHANGELOG.md:131]: I would move `### Changed` as the first section. This is the stuff that will contain breaking changes too and it feels pretty important.

File: tests/allow-privileged-import.earth [hunk 3]
```diff
@@ -1,8 +1,8 @@
-VERSION 0.6
+VERSION 0.7
 FROM alpine:3.15
 RUN apk add libcap # for capsh
 
-IMPORT --allow-privileged github.com/earthly/test-remote/privileged:main
+IMPORT --allow-privileged github.com/earthly/test-remote/privileged:45359a92acdd7ef27d561d0836099cdacc508e40
```
Comment (nelsam) [inline tests/allow-privileged-import.earth]: Will we need to update these tags after merge? (this is more of a process question than a review comment)

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,35 @@ All notable changes to [Earthly](https://github.com/earthly/earthly) will be doc
 
 ## Unreleased
 
+## v0.7.0-rc1 - 2022-10-12
```
Comment (vladaionescu) [inline CHANGELOG.md:7]: Did you mean to use a January date in here?

```suggestion
## v0.7.0-rc1 - 2023-01-13
```

Comment (vladaionescu) [inline CHANGELOG.md:7]: The release notes are not final, correct? Can we still edit them after merging to main? We will have more stuff in here to expand on. I would add most of the feature flags in more detail.

File: tests/remote-cache/Earthfile [hunk 5]
```diff
@@ -1,9 +1,28 @@
...
+--no-tar-build-output               \
+--shell-out-anywhere                \
+--use-cache-command                 \
+--use-chmod                         \
+--use-copy-link                     \
+--use-host-command                  \
+--use-no-manifest-list              \
+--use-pipelines                     \
+--use-project-secrets               \
+#--wait-block                        \
```
Comment (nelsam) [inline tests/remote-cache/Earthfile:19]: Was this comment intended to be left in?

Comment (nelsam) [reply tests/remote-cache/Earthfile:19]: Oh, I think this is the bug that we're fixing after the RC, right? got it.

Comment (alexcb) [reply tests/remote-cache/Earthfile:19]: This whole list comes from https://github.com/earthly/earthly/issues/2237#issuecomment-1340120452 (the features we're enabling for v0.7); I'm inclined to leave it commented out to both illustrate it's been disabled on purpose -- and to make it easier to re-enable when we loop back to fix this after the initial RC.

File: CHANGELOG.md (hunk 4 above)
Comment (alexcb) [reply CHANGELOG.md:7]: correct -- even though I'm merging this in -- I do not intend to release it as is.

**Review by vladaionescu — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (alexcb) [reply CHANGELOG.md:7]: We'll update this when we rewrite the release notes prior to the actual release -- there'll be a few more PRs before then.

File: tests/remote-cache/Earthfile (hunk 5 above)
Comment (nelsam) [reply tests/remote-cache/Earthfile:19]: Yeah, makes sense. I usually use `git revert` for those purposes, so I don't often leave the comments in - but it does serve as a good reminder.

**Review by nelsam — APPROVED** [review]

**Pull request merged**